driver-kotlin-extensions/src/test/kotlin/com/mongodb/kotlin/client/model/ProjectionTest.kt

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ * Copyright (C) 2016/2022 Litote
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @custom-license-header
+ */
+package com.mongodb.kotlin.client.model
+
+import com.mongodb.client.model.Aggregates.project
+import com.mongodb.client.model.Projections
+import com.mongodb.kotlin.client.model.Filters.and
+import com.mongodb.kotlin.client.model.Filters.eq
+import com.mongodb.kotlin.client.model.Filters.gt
+import kotlin.test.assertEquals
+import org.bson.BsonDocument
+import org.bson.conversions.Bson
+import org.junit.Test
+
+class ProjectionTest {
+
+    @Test
+    fun projection() {
+        assertEquals("\$name", Person::name.projection)
+        assertEquals("\$name.foo", Student::name.projectionWith("foo"))
+    }
+
+    @Test
+    fun include() {
+        assertEquals(
+            """{"name": 1, "age": 1, "results": 1, "address": 1}""",
+            include(Person::name, Person::age, Person::results, Person::address))
+
+        assertEquals("""{"name": 1, "age": 1}""", include(listOf(Person::name, Person::age)))
+        assertEquals("""{"name": 1, "age": 1}""", include(listOf(Person::name, Person::age, Person::name)))
+    }
+
+    @Test
+    fun exclude() {
+        assertEquals(
+            """{"name": 0, "age": 0, "results": 0, "address": 0}""",
+            exclude(Person::name, Person::age, Person::results, Person::address))
+        assertEquals("""{"name": 0, "age": 0}""", exclude(listOf(Person::name, Person::age)))
+        assertEquals("""{"name": 0, "age": 0}""", exclude(listOf(Person::name, Person::age, Person::name)))
+
+        assertEquals("""{"name": 0, "age": 0}""", exclude(listOf(Person::name, Person::age, Person::name)))
+
+        assertEquals("""{"_id": 0}""", excludeId())
+        assertEquals(
+            "Projections{projections=[{\"_id\": 0}, {\"name\": 1}]}",
+            fields(excludeId(), include(Person::name)).toString())
+    }
+
+    @Test
+    fun firstElem() {
+        assertEquals(""" {"name.${'$'}" : 1} """, Person::name.elemMatchProj)
+    }
+
+    @Test
+    fun elemMatch() {
+        // Given the following document:
```

Comment:
I get why you added this, but for me it reduces the readibility of the test cases. Especially as we're not testing the server and how it does projection. Also, this wasn't done for the Filters so that also adds the cognitive load.